modules/repository/create.go

```diff
@@ -207,6 +207,7 @@ func CreateRepository(doer, u *user_model.User, opts CreateRepoOptions) (*repo_m
 		IsEmpty:                         !opts.AutoInit,
 		TrustModel:                      opts.TrustModel,
 		IsMirror:                        opts.IsMirror,
+		DefaultBranch:                   opts.DefaultBranch,
```

Comment:
This is needed to use the returned `repo` with other methods without re-querying it from the database.